pkg/controller/jobframework/reconciler.go

```diff
@@ -1256,7 +1257,13 @@ func (r *JobReconciler) extractPriority(ctx context.Context, podSets []kueue.Pod
 	if jobWithPriorityClass, isImplemented := job.(JobWithPriorityClass); isImplemented {
 		customPriorityFunc = jobWithPriorityClass.PriorityClass
 	}
-	return ExtractPriority(ctx, r.client, job.Object(), podSets, customPriorityFunc)
+	priorityClassName, source, value, err := ExtractPriority(ctx, r.client, job.Object(), podSets, customPriorityFunc)
+	if apierrors.IsNotFound(err) {
+		r.record.Event(job.Object(), corev1.EventTypeWarning, ReasonPriorityNotFound, "PriorityClass not found")
```

Comment:
I think it would be great to include the name of the priority class and its source.